pinot-common/src/main/java/com/linkedin/pinot/pql/parsers/Pql2Compiler.java

```diff
@@ -55,6 +55,19 @@
 @ThreadSafe
 public class Pql2Compiler implements AbstractCompiler {
 
+  /**
+   * PQL compilers are not thread safe. Creating one per thread
```

Comment:
I think PQL compiler is now threadsafe, which is why the @ThreadSafe annotation was added at line 55.